gms/gossiper.cc

```diff
@@ -2326,6 +2327,20 @@ void gossiper::check_knows_remote_features(std::set<std::string_view>& local_fea
     }
 }
 
+void gossiper::check_snitch_name_matches(const std::unordered_set<gms::inet_address>& initial_contact_nodes, const std::unordered_set<gms::inet_address>& loaded_endpoints) {
+    for (const auto& contact_point : boost::range::join(initial_contact_nodes, loaded_endpoints)) {
```

Comment:
We need to check all the nodes in endpoint_state_map. For example, when a new node joins, the initial_contact_nodes can be a single node and loaded_endpoints is empty. endpoint_state_map contains all the nodes in the cluster. We should check them all.

Reply:
I updated it to use `endpoint_state_map`; @asias please have a look if this is what you asked for.

Reply:
I had one minor comment. Otherwise looks good to me. Thanks.